### PR TITLE
Update core to 10.0.0-rc.1

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [X] Updated ~Write tests for all new functionality. If tests were not written, please explain why.~
 - [X] Updated ~Add example if relevant.~
 - [X] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

- Updated:
  - MapboxCoreMaps 10.0.0-rc.1
  - MapboxCommon 14.0.1
  - Turf 2.0.0-beta.1
- Removed `CacheManager`. Functionality of some removed APIs will reintroduced in a later release.
- Changed `ResourceOptions.cachePathURL` to `dataPathURL` and removed `cacheSize`.

### CODE REVIEW

File: Apps/Examples/Examples/All Examples/OfflineManagerExample.swift [hunk 1]
```diff
@@ -52,20 +54,24 @@ public class OfflineManagerExample: UIViewController, ExampleProtocol {
...
-
+        logger = OfflineManagerLogWriter(textView: logView)
         state = .initial
     }
 
     // MARK: - Actions
 
     internal func downloadTileRegions() {
+        guard let tileStore = tileStore else {
+            preconditionFailure()
```
Comment (nishant-karajgikar) [inline Apps/Examples/Examples/All Examples/OfflineManagerExample.swift:72]: May want to log a message here, but can be done in a different pr..

File: Tests/MapboxMapsTests/MapView/Integration Tests/OfflineManagerIntegrationTests.swift [hunk 2]
```diff
@@ -28,16 +28,14 @@ internal class OfflineManagerIntegrationTestCase: IntegrationTestCase {
...
-        tileStorePathURL = try TileStore.fileURLForDirectory(for: name.fileSystemSafeString())
+        tileStorePathURL = try temporaryCacheDirectory()
 
-        // Ensure test starts from same conditions each time
-        try TileStore.removeDirectory(at: tileStorePathURL!)
-
-        tileStore = TileStore.shared(for: tileStorePathURL.path)
-        tileStore.setOptionForKey(TileStoreOptions.mapboxAccessToken, value: accessToken)
+        tileStore = TileStore.shared(for: tileStorePathURL)
+        tileStore.setOptionForKey(TileStoreOptions.mapboxAccessToken, value: accessToken as Any)
```
Comment (nishant-karajgikar) [inline Tests/MapboxMapsTests/MapView/Integration Tests/OfflineManagerIntegrationTests.swift:34]: I hadn't noticed this earlier.. It's a key-value store?

**Review by nishant-karajgikar — APPROVED** [review]

**Review by neelmistry94 — APPROVED** [review]

File: Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift [hunk 3]
```diff
@@ -88,41 +83,17 @@ public struct ResourceOptions {
...
-                cachePathURL: URL? = nil,
+                dataPathURL: URL? = nil,
                 assetPathURL: URL? = nil,
-                cacheSize: UInt64 = Self.defaultCacheSize,
                 tileStore: TileStore? = nil,
                 tileStoreUsageMode: TileStoreUsageMode = .readOnly) {
         self.accessToken        = accessToken
         self.baseURL            = baseURL
-        self.cachePathURL       = cachePathURL ?? ResourceOptions.cacheURLIncludingSubdirectory()
+        self.dataPathURL        = dataPathURL
```
Comment (pengdev) [inline Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift:92]: so the default is changed to nil? In Android we set the default path to /.mapbox/maps/ cc @julianrex

Comment (pengdev) [reply Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift:92]: I think the default gl-native will use is `/.mapbox/`

Comment (pengdev) [reply Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift:92]: We are going to pass null dataPath to core by default, creating an Android PR at https://github.com/mapbox/mapbox-maps-android/pull/403

**Pull request merged**